layer/logged_data/image.py

```diff
@@ -0,0 +1,121 @@
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Union
+
+import numpy as np
+import numpy.typing as npt
+
+from .utils import get_base_module_list, has_allowed_extension
+
+
+if TYPE_CHECKING:
+    import PIL
+    import torch
+
+
+class Image:
+    """
+    Helper class to log complex images such as torch.tensor or numpy.ndarray
+
+    Example of logging a numpy array as an image:
+    .. code-block:: python
+        ...
+        import numpy as np
+        img_HWC = np.zeros((100, 100, 3))
+        img_HWC[:, :, 0] = np.arange(0, 10000).reshape(100, 100) / 10000
+        img_HWC[:, :, 1] = 1 - np.arange(0, 10000).reshape(100, 100) / 10000
+
+        layer.log({"image":layer.Image(img_HWC)})
+        ...
+    """
+
+    def __init__(
+        self,
+        img: Union["PIL.Image.Image", Path, npt.NDArray[np.complex64], "torch.Tensor"],
+        format: str = "CHW",
+    ):
+        """
+        :param img: Supported image types are:
+        - torch.Tensor
+        - numpy.ndarray
+        - Image from path.Path
+        - PIL.Image
+        :param format: Sets the image format if the provided image is a numpy.ndarray.
+        Support formats are: `CHW`, `HWC`, `HW`
+        """
+        self.img = img
+        self.format = format
```

Comment:
`raise ValueError` if `format` is unsupported -- instead of doing it in `_get_image_from_array` below.